source/common/quic/http_datagram_handler.cc

```diff
@@ -0,0 +1,86 @@
+#include "source/common/quic/http_datagram_handler.h"
+
+#include "source/common/common/logger.h"
+#include "source/common/http/header_map_impl.h"
+
+#include "absl/strings/string_view.h"
+#include "quiche/common/simple_buffer_allocator.h"
+#include "quiche/quic/core/http/quic_spdy_stream.h"
+
+namespace Envoy {
+namespace Quic {
+
+HttpDatagramHandler::HttpDatagramHandler(quic::QuicSpdyStream* stream) : stream_(stream) {
+  ASSERT(stream != nullptr);
+  stream_->RegisterHttp3DatagramVisitor(this);
+}
+
+HttpDatagramHandler::~HttpDatagramHandler() {
+  if (stream_ != nullptr) {
```

Comment:
Is this ever false?

Reply:
It shouldn't be. Thanks for the suggestion of using reference instead of pointer.